frontend/src/metabase/redux/embed/embed.ts

```diff
@@ -0,0 +1,154 @@
+import { type PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { compose, pick } from "underscore";
+
+import { DEFAULT_EMBEDDING_ENTITY_TYPES } from "metabase/embedding-sdk/store";
+import { parseSearchOptions } from "metabase/lib/browser";
+import type { InteractiveEmbeddingOptions } from "metabase-types/store";
+
+export const DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS: InteractiveEmbeddingOptions =
+  {
+    font: undefined,
+    top_nav: true,
+    side_nav: "default",
+    search: false,
+    new_button: false,
+    breadcrumbs: true,
+    logo: true,
+    header: true,
+    additional_info: true,
+    action_buttons: true,
+    entity_types: DEFAULT_EMBEDDING_ENTITY_TYPES,
+  };
+
+const ALLOWED_INTERACTIVE_EMBEDDING_OPTIONS = Object.keys(
+  DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
+);
+
+export const urlParameterToBoolean = (
+  urlParameter: string | string[] | boolean | undefined,
+) => {
+  if (urlParameter === undefined) {
+    return undefined;
+  }
+  if (Array.isArray(urlParameter)) {
+    return Boolean(urlParameter.at(-1));
+  } else {
+    return Boolean(urlParameter);
+  }
+};
+
+const interactiveEmbedSlice = createSlice({
+  name: "interactiveEmbed",
+  initialState: {
+    options: {} as InteractiveEmbeddingOptions,
+    isEmbeddingSdk: false,
+  },
+  reducers: {
+    setInitialUrlOptions: (
+      state,
+      action: PayloadAction<{ search: string }>,
+    ) => {
+      const searchOptions = compose(
+        normalizeEntityTypes,
+        excludeNonInteractiveEmbeddingOptions,
+        parseSearchOptions,
+        normalizeCommaSeparatedSearchOptions(["entity_types"]),
+      )(action.payload.search);
+
+      state.options = {
+        ...DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
+        ...searchOptions,
+      };
+    },
+    setOptions: (
+      state,
+      action: PayloadAction<Partial<InteractiveEmbeddingOptions>>,
+    ) => {
+      state.options = {
+        ...state.options,
+        ...action.payload,
+      };
+    },
+  },
+});
+
+/**
+ * this functions turns a string like `param=value1,value2` into `param=value1&param=value2` that matches the URLSearchParams format
```

Comment:
Nit: from reading the function definition that's not really what it does. It only does that for the keys passed as through `options`. Maybe the comment should be clearer?

Additionally, this function seems to only be called once which a specific argument, maybe it doesn't need to be this generic?

Since you are excluding non-option type later on with `excludeNonInteractiveEmbeddingOptions` anyway, maybe this function does not need an input at all and can just normalize all params in the search string?

Reply:
addressed in 0977fdd97fbdb7070b8ccadf1f7681b1e31fc7b9